File: LeetCode Solutions/March 2022 LeetCoding Challenges/Search a 2D Matrix/Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row, column = -1, len(matrix[0])
        for i in range(len(matrix)):
            if target <= matrix[i][column-1]:
                row = i
                break
        if row == -1:
            return False
        for j in range(len(matrix[0])):
            if matrix[row][j] == target:
                return True
        return False

File: LeetCode Solutions/March 2022 LeetCoding Challenge/Search a 2D Matrix/Search a 2D Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int row = -1;
        int column = matrix[0].length;
        for(int i=0; i<matrix.length; i++){
            if(target <= matrix[i][column-1]){
                row = i;
                break;
            }
        }
        if(row == -1){
            return false;
        }
        for(int j=0; j<matrix[0].length; j++){
            if(matrix[row][j] == target){
                return true;
            }
        }
        return false;
    }
}